overlord/snapstate/backend/link.go

```diff
@@ -184,6 +185,35 @@ func (b Backend) LinkSnap(info *snap.Info, dev snap.Device, linkCtx LinkContext,
 	return rebootInfo, nil
 }
 
+func componentLinkPath(cpi snap.ContainerPlaceInfo, snapRev snap.Revision) string {
+	instanceName, compName, _ := strings.Cut(cpi.ContainerName(), "+")
+	compBase := snap.ComponentsBaseDir(instanceName)
+	return filepath.Join(compBase, snapRev.String(), compName)
+}
+
+func (b Backend) LinkComponent(cpi snap.ContainerPlaceInfo, snapRev snap.Revision) error {
+	mountDir := cpi.MountDir()
+	linkPath := componentLinkPath(cpi, snapRev)
+
+	// We want a relative symlink, so we remove the common parts of the
+	// path and then go up the needed directories.
+	linkComps := strings.Split(linkPath, "/")
```

Comment:
would `filepath.Rel()` work here?

Reply:
Right, it makes this easier, changed now.